File: app/src/main/java/com/beone/bestpractice/local/model/CountryEntity.java
package com.beone.bestpractice.local.model;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.room.Embedded;
import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

@Entity(tableName = "countryentity")
public class CountryEntity {

  @Expose
  @SerializedName("critical")
  private int critical;
  @Expose
  @SerializedName("active")
  private int active;
  @Expose
  @SerializedName("recovered")
  private int recovered;
  @Expose
  @SerializedName("todayDeaths")
  private int todayDeaths;
  @Expose
  @SerializedName("deaths")
  private int deaths;
  @Expose
  @SerializedName("todayCases")
  private int todayCases;
  @Expose
  @SerializedName("cases")
  private int cases;

  @Expose
  @SerializedName("tests")
  private int tests;



  @Expose
  @SerializedName("countryInfo")
  @Embedded
  private CountryInfo countryInfo;
  @Expose
  @SerializedName("country")
  @PrimaryKey(autoGenerate = false)
  @NonNull
  private String country;

  public int getCritical() {
    return critical;
  }

  public int getActive() {
    return active;
  }

  public int getRecovered() {
    return recovered;
  }

  public int getTodayDeaths() {
    return todayDeaths;
  }

  public int getDeaths() {
    return deaths;
  }

  public int getTodayCases() {
    return todayCases;
  }

  public int getCases() {
    return cases;
  }

  public CountryInfo getCountryInfo() {
    return countryInfo;
  }

  public String getCountry() {
    return country;
  }

  public int getTests() {
    return tests;
  }

  public void setTests(int tests) {
    this.tests = tests;
  }

  public void setCritical(int critical) {
    this.critical = critical;
  }

  public void setActive(int active) {
    this.active = active;
  }

  public void setRecovered(int recovered) {
    this.recovered = recovered;
  }

  public void setTodayDeaths(int todayDeaths) {
    this.todayDeaths = todayDeaths;
  }

  public void setDeaths(int deaths) {
    this.deaths = deaths;
  }

  public void setTodayCases(int todayCases) {
    this.todayCases = todayCases;
  }

  public void setCases(int cases) {
    this.cases = cases;
  }

  public void setCountryInfo(CountryInfo countryInfo) {
    this.countryInfo = countryInfo;
  }

  public void setCountry(String country) {
    this.country = country;
  }
}

File: app/src/main/java/com/beone/bestpractice/core/data/source/local/entities/CountryEntity.kt
package com.beone.bestpractice.core.data.local.entities

import android.os.Parcelable
import androidx.room.Embedded
import androidx.room.Entity
import androidx.room.PrimaryKey
import kotlinx.android.parcel.Parcelize
import kotlinx.android.parcel.RawValue
import org.jetbrains.annotations.Nullable

@Parcelize
@Entity(tableName = "country")
data class CountryEntity(
        @PrimaryKey(autoGenerate = true)
        @Nullable
        val countryId: Int?,
        val active: Int,
        val activePerOneMillion: Double,
        val cases: Int,
        val casesPerOneMillion: Int,
        val continent: String,
        val country: String,
        val critical: Int,
        val criticalPerOneMillion: Double,
        val deaths: Int,
        val deathsPerOneMillion: Double,
        val oneCasePerPeople: Int,
        val oneDeathPerPeople: Int,
        val oneTestPerPeople: Int,
        val population: Int,
        val recovered: Int,
        val recoveredPerOneMillion: Double,
        val tests: Int,
        val testsPerOneMillion: Int,
        val todayCases: Int,
        val todayDeaths: Int,
        val todayRecovered: Int,
        val updated: Long,
        @Embedded val countryInfoEntity: CountryInfoEntity
):Parcelable

@Parcelize
data class CountryInfoEntity(
        var _id: Int = 0,
        var flag: String ="",
         var iso2: String ="",
         var iso3: String ="",
        var lat: Double =0.0,
        var long: Double=0.0
):Parcelable